IBM/ubiquity-k8s
# ubiquity-db pv name as configurable parameter for the installer (to support DS8K)

Change:
    In ubiquity 1.0.0, ibm-ubiquity-db is a hardcord pv name, but to support DS8k which has volume name length limitation about 16 chars, so we change design about ubiquity db name, make it configurable in ubiquity_installer.conf and ubiquity-configmap.yml, so customer can set the default name "ibm-ubiquity-db" to example "ibmdb" if the backend storage is DS8k

How to test:
basic testing:
   1. The default db name works on SVC/XIV, and not works on DS8k
   2. Change default db name to "ibmdb" and make sure it works on DS8k
   3. Change db name to "ibm-ubiquity-db-test", and it should also works on SVC/XIV
   4. Verify uninstall successfully

upgrade testing:
  steps:
        Had ubiquity 1.0.0 installed, backend storage is SVC/XIV
        Upgrade ubiquity-configmap.yml with ubiquity_installer.conf(use default db name)
        Upgrade configmap with new ubiquity-configmap yml
        Upgrade Ubiquity and Provisioner Pod
        Make sure everything works fine

Documents:
     1.  ibm-ubiquity-db is the default db name for SVC/XIV, for DS8k, we suggest to use ibmdb, also customer should know that db name: u_{instance}_ibmdb should less than 17 chars
     2. if customer want to create pv in DS8k, customer need to set pv name with pv-name label in pvc, otherwize it will fail to create pvc because DS8k has 16 chars limitation
     3. Need to tell customer how to do upgrade







Signed-off-by: feihuang <feihuang@feihuangs-mbp.cn.ibm.com>


---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/ibm/ubiquity-k8s/167)
